src/core/control/tools/EllipseHandler.cpp

```diff
@@ -63,8 +63,16 @@ auto EllipseHandler::createShape(bool isAltDown, bool isShiftDown, bool isContro
     /*
      * Set resolution depending on the radius (heuristic)
      */
-    auto nbPtsPerQuadrant =
-            static_cast<unsigned int>(std::ceil(5 + 0.2 * std::max(std::abs(radiusX), std::abs(radiusY))));
+    unsigned int nbPtsPerQuadrant;
+
+    const double excentricity = std::abs(radiusY / radiusX);
```

Comment:
I see no need to separate cases by `excentricity`, since rx + ry is an approximation to max(rx,ry) if either rx or ry is extremely smaller than the other. @bhennion's formula is good for all cases.